MapboxNavigation/Style.swift

```diff
@@ -209,6 +210,95 @@ public class ResumeButton: UIControl {
     }
 }
 
+/// :nodoc:
+@objc(MBSpeedLimitSign)
+open class SpeedLimitSign: UIView {
+    
+    private static let textInsets = UIEdgeInsets(top: 6, left: 6, bottom: 6, right: 6)
+    
+    lazy var label: SpeedLimitLabel = .forAutoLayout()
+    
+    let defaultText = NSLocalizedString("SPEED_LIMIT", bundle: .mapboxNavigation, value: "Speed\nLimit\n", comment: "Speed limit sign main text.")
+    let paragraphStyle = NSMutableParagraphStyle()
+    
+    var speedLimit: SpeedLimit? {
+        didSet {
+            guard let speedLimit = speedLimit else { return }
+            let fontSize = UIFont.systemFontSize
+            
+            let strings = NSMutableAttributedString()
+            
+            strings.append(NSMutableAttributedString(string: defaultText.uppercased(), attributes: [
+                .paragraphStyle: paragraphStyle,
+                .font: UIFont.systemFont(ofSize: fontSize * 0.9)
+                ]))
+            
+            strings.append(NSMutableAttributedString(string: String(describing: speedLimit.speed), attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 2),
+                .paragraphStyle: paragraphStyle
+                ]))
+            
+            strings.append(NSMutableAttributedString(string: "\n\(speedLimit.speedUnits.localizedSpeedUnit)", attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize / 1.3),
+                .paragraphStyle: paragraphStyle
+                ]))
+            
+            label.attributedText = strings
+        }
+    }
+    
+    @objc dynamic public var borderColor: UIColor? {
+        get {
+            guard let color = layer.borderColor else { return nil }
+            return UIColor(cgColor: color)
+        }
+        set {
+            layer.borderColor = newValue?.cgColor
+        }
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    
+    func commonInit() {
+        addSubview(label)
+        paragraphStyle.lineHeightMultiple = 0.9
+        paragraphStyle.alignment = .center
+        layoutMargins = SpeedLimitSign.textInsets
+        label.lineBreakMode = .byWordWrapping
+        label.numberOfLines = 4
```

Comment:
This skeuomorphic design attempts to reproduce the [MUTCD](https://en.wikipedia.org/wiki/Manual_on_Uniform_Traffic_Control_Devices)-style [speed limit sign](https://en.wikipedia.org/wiki/Speed_limit#Maximum_speed_limit) used in the 50 U.S. states and (with modifications) in Canada and Puerto Rico. This is very different than the design used in some form or another everywhere else in the world.

<img src="https://upload.wikimedia.org/wikipedia/commons/d/de/Hungary_road_sign_C-033-50.svg" width="100" alt="50 km/h">

The plan is for the Directions API to indicate which sign design to use (or which country the step is in), allowing for this SDK to switch between different layouts.

Shorter-term, if the API ends up supporting speed limit attributes in Europe first, then we’ll need to rework this code to unconditionally display a Vienna Convention–style sign.